ci/utils.sc

```diff
@@ -94,6 +93,20 @@ def runWithTimeout(timeout: FiniteDuration, logFileName: String)(commands: Seq[S
       val cmd = commands.mkString(" ")
       throw new java.util.concurrent.TimeoutException(s"'$cmd' timed out after $timeout.")
     }
+}
+
+/**
+ * Run a process with given commands and time out it runs too long.
+ *
+ * @param timeout The maximum time to wait.
+ * @param logFileName Name of file which collects all logs.
+ * @param commands The commands that are executed in a process. E.g. "sbt",
+ *  "compile".
+ */
+def runWithTimeoutAndCleanup(timeout: FiniteDuration, logFileName: String)(commands: Seq[String]): Unit = {
```

Comment:
What is the motivation for this?

Reply:
because the call for flake8 is a python thing that can't possibly need Java process cleanup.  It was likely called with it out of convenience to reuse what was there and gain the benefits of a timeout... 
this is better.

Reply:
Ok, makes sense. How about this

```scala
def runWithTimeout(timeout: FiniteDuration, logFileName: String)(commands: Seq[String]): Unit = ???

def withCleanUp[T](block :=> T): T {
  try { block }
  finally { provision.killStaleTestProcesses() }
}

def run(cmd: String *) = withCleanUp { utils.runWithTimeoutAndCleanup(30.minutes, logFileName)(cmd) }
```